### PR TITLE
[REVIEW] Pin gdal to 3.2.x

### PR DESCRIPTION
**Pull request by weiji14**
Bump `gdal` from 3.1.x to 3.2.x. Supersedes #339. This is inline with conda-forge's current migration to gdal=3.2.x (see https://github.com/conda-forge/conda-forge-pinning-feedstock/pull/922 and https://conda-forge.org/status/#gdal32).

Update on rapidsai/integration is at https://github.com/rapidsai/integration/pull/211.

### CODE REVIEW

**Comment by GPUtester** [comment]
Can one of the admins verify this patch?

**Comment by GPUtester** [comment]
Can one of the admins verify this patch?

**Comment by GPUtester** [comment]
Can one of the admins verify this patch?

**Comment by ajschmidt8** [comment]
ok to test

**Review by kkraus14 — APPROVED** [review]

File: ci/gpu/build.sh [hunk 1]
```diff
@@ -43,9 +43,9 @@ conda activate rapids
 gpuci_conda_retry install "cudf=${MINOR_VERSION}.*" "cudatoolkit=$CUDA_REL" \
     "rapids-build-env=$MINOR_VERSION.*"
 
-# https://docs.rapids.ai/maintainers/depmgmt/ 
-# gpuci_conda_retry remove --force rapids-build-env
-# gpuci_conda_retry install "your-pkg=1.0.0"
+# https://docs.rapids.ai/maintainers/depmgmt/
+gpuci_conda_retry remove --force rapids-build-env
+gpuci_conda_retry install "gdal=3.2.1"
```
Comment (weiji14) [inline ci/gpu/build.sh:48]: Reminder to revert changes to these lines once tests pass.

```suggestion
# https://docs.rapids.ai/maintainers/depmgmt/ 
# gpuci_conda_retry remove --force rapids-build-env
# gpuci_conda_retry install "your-pkg=1.0.0"
```

**Comment by ajschmidt8** [comment]
@weiji14, we just merged https://github.com/rapidsai/integration/pull/211. if you revert the temporary testing changes in your PR here, we can merge this as well :+1:

**Review by ajschmidt8 — APPROVED** [review]

**Comment by ajschmidt8** [comment]
CI previously passed. Current failures are due to `gdal`  version mismatch which is resolved in https://github.com/rapidsai/integration/pull/211.

Manually merging this PR.

**Pull request merged**